analysis/known_objects.py

```diff
@@ -0,0 +1,361 @@
+from astropy.coordinates import Angle
+from astropy.coordinates import SkyCoord
+from astropy.time import Time
+import astropy.units as u
+
+from astroquery.imcce import Skybot
+from astroquery.jplhorizons import Horizons
+from image_info import *
+
+import json
+import urllib.request as libreq
+
+class KnownObjects():
+    def __init__(self):
+        self.objects = {}
+        self.max_time_index = -1
+        self.times = []
+        
+    def get_num_times(self):
+        return self.max_time_index + 1
+    
+    def get_num_results(self):
+        return len(self.objects.keys())
+    
+    def set_timestamp(self, time_index, t):
+        while len(self.times) <= time_index:
+            self.times.append(0.0)
+        self.times[time_index] = t
+
+    def add_observation(self, name, time_index, sc):
+        """
+        Add an observation to the set seen known objects.
+        
+        Arguments:
+            name : string
+                Object name from the databse.
+            time_index : integer
+                The index of the timestep for this observation.
+            sc : a SkyCoordinate object 
+                Indicates the position of the object.
+        """
+        if self.max_time_index < time_index:
+            self.max_time_index = time_index
+
+        # If we haven't seen this object before add an array
+        # to store the observations.
+        if name not in self.objects:
+            self.objects[name] = []
+
+        # Fill in any missing observations with None.
+        while len(self.objects[name]) < time_index - 1:
+            self.objects[name].append(None)
+
+        # Add the new observation.
+        self.objects[name].append(sc)
+
+    def get_num_times_seen(self, name):
+        """
+        Count the number of times we saw an object.
+        
+        Arguments:
+            name : string
+                Object name from the databse.
+                
+        Returns:
+            integer : the count of times we should have
+                      seen this object.
+        """
+        if not name in self.objects:
+            return 0
+
+        count = 0
+        for sc in self.objects[name]:
+            if sc is not None:
+                count = count + 1
+        return count
+
+    def add_observation(self, name, time_index, sc):
```

Comment:
This definition looks like a duplicate of the above add_observation.

Reply:
Again... not sure how that happened. I removed the duplicate.